pkg/queue/requeue_intervals.go

```diff
@@ -11,23 +12,58 @@ type RequeueIntervals struct {
 	Busy    time.Duration
 	Warning time.Duration
 	Error   time.Duration
+	Jitter  *RequeueJitter
 }
 
 func DetermineRequeueInterval(state shared.State, intervals RequeueIntervals) time.Duration {
+	var interval time.Duration
 	switch state {
 	case shared.StateError:
-		return intervals.Error
+		interval = intervals.Error
 	case shared.StateDeleting:
-		return intervals.Busy
+		interval = intervals.Busy
 	case shared.StateProcessing:
-		return intervals.Busy
+		interval = intervals.Busy
 	case shared.StateWarning:
-		return intervals.Warning
+		interval = intervals.Warning
 	case shared.StateReady:
-		return intervals.Success
+		interval = intervals.Success
 	default:
-		return intervals.Success
+		interval = intervals.Success
 	}
+
+	if intervals.Jitter != nil {
+		return intervals.Jitter.Apply(interval)
+	}
+	return interval
+}
+
+type RequeueJitter struct {
+	Enabled           bool
+	StartedAt         time.Time
+	DisableAfter      time.Duration
+	JitterProbability float64
+	JitterPercentage  float64
+}
+
+func (j *RequeueJitter) Apply(interval time.Duration) time.Duration {
+	if !j.Enabled {
+		return interval
+	}
+
+	if time.Since(j.StartedAt) > j.DisableAfter {
+		j.Enabled = false
+		return interval
+	}
+
+	//nolint:gosec // No simple way to generate a float in crypto/rand lib and this is not a security-sensitive context
+	if rand.Float64() < j.JitterProbability {
```

Comment:
That looks OK when I look at it, please provide an unit test for it though.
In order to test it you externalize `rand.Float64` into a new field in `RequeueJitter` struct, say, `randFunc func() float64`
Then in your test you can configure the `RequeueJitter` struct with a function that always returns `0.02` for example.
In the "real" code you wire-up math.Rand() to this field (main.go: 407)
We don't have to "test" math.Rand() directly, it's assumed to be working fine.